src/db/users.rs

```diff
@@ -5,6 +5,22 @@ use diesel;
 use diesel::pg::PgConnection;
 use diesel::prelude::*;
 use serde::Deserialize;
+use diesel::dsl::{exists, select};
+
+pub fn username_exists(conn: &PgConnection, _username: &str) -> bool {
+    use crate::schema::users::dsl::*;
+    let username_exists = select(exists(users.filter(username.eq(_username))))
```

Comment:
This temp variable doesn't seem to be really useful. Should the result of the selection be returned immediately?

Reply:
I assumed that for readability it was okay to put it, but it is certainly enough with the name of the function to specify what the body does, I will directly return the select

Reply:
Another thing: `_username` is a typical syntax for unused variables.
I can see 2 alternatives:
- rename argument
- or something like this:
```rust
    use crate::schema::users;
    let username_exists = select(exists(users::tables.filter(users::username.eq(username))))
```

Reply:
Oh, ok, the rust compiler show a warning, the suggestion is to use # [warn (unused_variables)]  but that's too much I suppose, basically it's a compiler bug, that's why the underscore



warning: unused variable: `username`
  --> src/db/users.rs:10:45
   |
10 | pub fn username_exists(conn: &PgConnection, username: &str) -> bool {
   |                                             ^^^^^^^^ help: consider prefixing with an underscore: `_username`
   |
   = note: #[warn(unused_variables)] on by default